zebrad/tests/acceptance.rs

```diff
@@ -461,3 +461,37 @@ fn valid_generated_config(command: &str, expected_output: &str) -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn metrics_tracing_listeners() -> Result<()> {
+    zebra_test::init();
+
+    // Write a configuration that has both metrics endpoint_addr and tracing endpoint_addr options set
+    let mut config = default_test_config()?;
+    config.metrics.endpoint_addr = Some("127.0.0.1:0".parse().unwrap());
+    config.tracing.endpoint_addr = Some("127.0.0.1:0".parse().unwrap());
+
+    let dir = TempDir::new("zebrad_tests")?;
+    fs::File::create(dir.path().join("zebrad.toml"))?
+        .write_all(toml::to_string(&config)?.as_bytes())?;
+
+    let tempdir = dir.path().to_path_buf();
+
+    let mut child = get_child(&["start"], &tempdir)?;
+
+    // Run the program and kill it at 1 second
+    std::thread::sleep(Duration::from_secs(1));
+    child.kill()?;
```

Comment:
before we kill the process, could we try connecting to each endpoint and do some basic interactions with them like setting the filter and then looking it up.

Reply:
What do you think it will be the best crate to be used for connecting to the endpoints ? `std::net::TcpStream` was the natural choice for me however not totally sure, there are other options.

Reply:
I think we already have a dependency on `hyper` so that would probably be better, I think doing it via `TcpStream` directly might force you to deal with http headers, not sure though.

Reply:
`hyper` looks good to me. Thanks.

Reply:
Applied partially at https://github.com/ZcashFoundation/zebra/pull/1000/commits/21387745f5729d63cffb1b2b23eafa503e30d11b

Set the filter is blocked by https://github.com/ZcashFoundation/zebra/issues/1001